e2e/cypress/integration/api/iam/08_projects_api_nodemanager_update.ts

```diff
@@ -0,0 +1,288 @@
+        name: 'project environment'
+      },
+      rule: {
+        id: 'rule-environment',
+        name: 'rule ENVIRONMENT',
+        type: 'NODE',
+        project_id: `${cypressPrefix}-project-environment-${Cypress.moment().format('MMDDYYhhmm')}`,
+        conditions: [
+          {
+            attribute: 'ENVIRONMENT',
+            operator: 'EQUALS',
+            values: ['arctic']
+          }
+        ]
+      }
+    },
+    {
+      project: {
+        id: `${cypressPrefix}-project-policy-group-${Cypress.moment().format('MMDDYYhhmm')}`,
+        name: 'project policy group'
+      },
+      rule: {
+        id: 'rule-policy-group',
+        name: 'rule CHEF_POLICY_GROUP',
+        type: 'NODE',
+        project_id:
+        `${cypressPrefix}-project-policy-group-${Cypress.moment().format('MMDDYYhhmm')}`,
+        conditions: [
+          {
+            attribute: 'CHEF_POLICY_GROUP',
+            operator: 'EQUALS',
+            values: ['red_ring']
+          }
+        ]
+      }
+    },
+    {
+      project: {
+        id: `${cypressPrefix}-project-policy-name-${Cypress.moment().format('MMDDYYhhmm')}`,
+        name: 'project policy name'
+      },
+      rule: {
+        id: 'rule-policy-name',
+        name: 'rule CHEF_POLICY_NAME',
+        type: 'NODE',
+        project_id: `${cypressPrefix}-project-policy-name-${Cypress.moment().format('MMDDYYhhmm')}`,
+        conditions: [
+          {
+            attribute: 'CHEF_POLICY_NAME',
+            operator: 'EQUALS',
+            values: ['fire']
+          }
+        ]
+      }
+    },
+    {
+      project: {
+        id: `${cypressPrefix}-project-role-${Cypress.moment().format('MMDDYYhhmm')}`,
+        name: 'project role'
+      },
+      rule: {
+        id: 'rule-role',
+        name: 'rule CHEF_ROLE',
+        type: 'NODE',
+        project_id: `${cypressPrefix}-project-role-${Cypress.moment().format('MMDDYYhhmm')}`,
+        conditions: [
+          {
+            attribute: 'CHEF_ROLE',
+            operator: 'EQUALS',
+            values: ['backend']
+          }
+        ]
+      }
+    },
+    {
+      project: {
+        id: `${cypressPrefix}-project-tag-${Cypress.moment().format('MMDDYYhhmm')}`,
+        name: 'project tag'
+      },
+      rule: {
+        id: 'rule-tag',
+        name: 'rule CHEF_TAG',
+        type: 'NODE',
+        project_id: `${cypressPrefix}-project-tag-${Cypress.moment().format('MMDDYYhhmm')}`,
+        conditions: [
+          {
+            attribute: 'CHEF_TAG',
+            operator: 'EQUALS',
+            values: ['v3']
+          }
+        ]
+      }
+    }
+  ];
+
+  before(() => {
+    cy.cleanupV2IAMObjectsByIDPrefixes(cypressPrefix, ['projects', 'policies']);
+  });
+
+  after(() => cy.cleanupV2IAMObjectsByIDPrefixes(cypressPrefix, ['projects', 'policies']));
+
+  it('tagging 7 projects with all the attributes on a compliance and client runs node', () => {
```

Comment:
I'd second reorganizing the test cases into a map. Then we could have more specific test names, like `when a project has a rule that matches one compliance node's and one client runs node's ${attribute} attribute, successfully associates those nodes with that project`

Reply:
I would like that too, but I could not think of a way to do that and only call "apply rules" once.

Reply:
🤔lemme ponder that

Reply:
what I ended up doing in [this PR](https://github.com/chef/automate/pull/2322/files#diff-e1765dc0e83555d8980cd1d444bf612cR103) is running most of the actual API calls in the `before` block. then the test themselves just looped over each projectRule to assert the action got tagged correctly. it's a weird structure but I like having more specific test output

Reply:
Yeah, I kind of did the same thing. Take a look and tell me if you think this is better.